helpers/social_connection_helper.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import re
+
+from consts.social_connection_type import SocialConnectionType
+
+
+class SocialConnectionHelper(object):
+    @classmethod
+    def group_by_type(cls, connections):
+        conns_by_type = {}
+        for conn in connections:
+            slug_name = conn.slug_name
+            if slug_name in conns_by_type:
+                conns_by_type[slug_name].append(conn)
+            else:
+                conns_by_type[slug_name] = [conn]
+        return conns_by_type
+
+
+class SocialConnectionParser(object):
+    FACEBOOK_URL_PATTERNS = ['facebook.com/']
+    TWITTER_URL_PATTERNS = ['twitter.com/']
+    YOUTUBE_URL_PATTERNS = ['youtube.com/user/']
+    GITHUB_URL_PATTERNS = ['github.com/']
+
+    @classmethod
+    def partial_social_dict_from_url(cls, url):
+        """
+        Takes a url, and turns it into a partial Media object dict
+        """
+        if any(s in url for s in cls.FACEBOOK_URL_PATTERNS):
```

Comment:
Is there a way to put all the patterns in a list (maybe initially have them all in a dict?) and iterate through it? This feels repetitive. It seems like you'd need to touch code in a lot of places to add a new pattern.